resources/todomvc/architecture-examples/preact/src/app/item.js

```diff
@@ -1,85 +1,60 @@
-import cx from 'classnames';
-import { h, Component } from 'preact';
-
-const ESCAPE_KEY = 27;
-const ENTER_KEY = 13;
-
-export default class TodoItem extends Component {
-    handleSubmit = () => {
-        let { onSave, onDestroy, todo } = this.props,
-            val = this.state.editText.trim();
+import cx from "classnames";
+// eslint-disable-next-line no-unused-vars
+import { h } from "preact";
+import { useState, useEffect } from "preact/hooks";
+export default function TodoItem({ onSave, onRemove, onToggle, todo }) {
+    const [editing, setEditing] = useState(false);
+
+
+    /**
+     * useEffect keeps track of the 'editing' state change.
+     * If the input field is present, we set focus programmatically.
+     */
+    useEffect(() => {
+        const input = document.querySelector(".editing");
```

Comment:
@julienw - I tested using useRef and it increased the time. 
This works, but it's not recommended maybe? 

We can either:
1. useRef and be ok with the increased time
2. keep this